src/ApiClient.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  Cursor,
+  Envelope,
+  MessageApi,
+  PagingInfo,
+  PublishRequest,
+  QueryRequest,
+  SortDirection,
+  SubscribeRequest,
+} from '@xmtp/proto'
+import { retry } from './utils'
+import { NotifyStreamEntityArrival } from '@xmtp/proto/ts/dist/types/fetch.pb'
+
+const RETRY_SLEEP_TIME = 100
+
+export type QueryParams = {
+  startTime?: Date
+  endTime?: Date
+  contentTopics: string[]
+}
+
+export type QueryAllOptions = {
+  direction?: SortDirection
+  limit?: number
+}
+
+export type QueryStreamOptions = Omit<QueryAllOptions, 'limit'> & {
+  pageSize?: number
```

Comment:
Should we just call this `limit` too?

Reply:
I wanted to make it a little more explicit that you would still receive 100% of the items that match the query in this case.